### PR TITLE
fix: close notification with Enter key

### LINKED ISSUE #299: Unable to close notification using the Enter key
### Reproduction link

[Edit on CodeSandbox](https://codesandbox.io/s/late-worker-e3loql?file=/src/App.tsx)

### Steps to reproduce

1. Load the sandbox
2. Click `Show notification`
3. See that the close icon gets focus and turns red
4. See that the notification is not closed after 1s (simulated keyboard event via timeout)
6. (Optional) Simulate the click event and see that the notification gets closed after 1s

### What is expected?

When close button has keyboard focus, pressing Enter on the keyboard should close the notification.

### What is actually happening?

When close button has keyboard focus, pressing Enter on the keyboard does not close the notification.

| Environment | Info |
| --- | --- |
| rc-notification | 4.6.0 |
| React | 18 |
| System | ALL |
| Browser | ALL |

---

Ref: UIEN-1752

### PR DESCRIPTION
**Pull request by MehmetYararVX**
fixes #299

### CODE REVIEW

File: src/Notice.tsx [hunk 1]
```diff
@@ -1,5 +1,4 @@
 import * as React from 'react';
-import classNames from 'classnames';
```
[the remark this answers is not in the document]
Comment (zombieJ) [reply src/Notice.tsx:2]: Seems bug, you can add this back with `git commit -n` to tmp resolve this.

File: src/Notice.tsx [hunk 2]
```diff
@@ -49,6 +50,14 @@ const Notify = React.forwardRef<HTMLDivElement, NoticeProps & { times?: number }
     onNoticeClose(eventKey);
   };
 
+  const handleCloseKeydown: React.KeyboardEventHandler<HTMLAnchorElement> = closable
```
Comment (zombieJ) [inline src/Notice.tsx:53]: `handleCloseKeydown` -> `onCloseKeyDown`

Comment (zombieJ) [reply src/Notice.tsx:53]: no need `closable` check since close btn will not show when `closable` is `false`.

**Comment by mellis481** [comment]
@zombieJ Can you advise why the close element is an anchor rather than a button? Buttons are to be used for performing an in-page action and anchors/links are for navigating to a new page. Based off this, the close element should certainly be a button. Should that be done in another PR?

**Comment by zombieJ** [comment]
This is a historical legacy, and now antd, which relies on rc, no longer supports legacy browsers. It is indeed possible to open a new PR to change it.

**Comment by zombieJ** [comment]
`+ rc-notification@5.0.3`

**Comment by mellis481** [comment]
@zombieJ Thanks for merging and releasing this important PR. Will this get updated in antd soon or should we create a PR for that now?

**Comment by yoyo837** [comment]
Remove `node_modules` and lock file, then reinstall it.

**Pull request merged**